Patch-Source: https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1770
---
From e825638a2ae218e6ef9c7a100509f7095cd2ee29 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 6 Aug 2025 18:06:31 +0200
Subject: [PATCH 1/2] cell-broadcast-prompt: Close dialog on swipe
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes: b281cc5d3 ("cell-broadcast-prompt: New dialog for cell broadcast message")
Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1753>
(cherry picked from commit 17453a8d55b80aa83c09ff6eba67a1408c5f99c9)
---
 src/cell-broadcast-prompt.c     | 8 ++++++++
 src/ui/cell-broadcast-prompt.ui | 1 +
 2 files changed, 9 insertions(+)

diff --git a/src/cell-broadcast-prompt.c b/src/cell-broadcast-prompt.c
index 7336db0bc..78ecf04e7 100644
--- a/src/cell-broadcast-prompt.c
+++ b/src/cell-broadcast-prompt.c
@@ -112,6 +112,13 @@ on_ok_clicked (PhoshCellBroadcastPrompt *self)
 }
 
 
+static void
+on_dialog_canceled (PhoshCellBroadcastPrompt *self)
+{
+  g_signal_emit (self, signals[CLOSED], 0);
+}
+
+
 static void
 phosh_cell_broadcast_prompt_finalize (GObject *obj)
 {
@@ -147,6 +154,7 @@ phosh_cell_broadcast_prompt_class_init (PhoshCellBroadcastPromptClass *klass)
   gtk_widget_class_set_template_from_resource (widget_class,
                                                "/mobi/phosh/ui/cell-broadcast-prompt.ui");
   gtk_widget_class_bind_template_child (widget_class, PhoshCellBroadcastPrompt, lbl_msg);
+  gtk_widget_class_bind_template_callback (widget_class, on_dialog_canceled);
   gtk_widget_class_bind_template_callback (widget_class, on_ok_clicked);
 
   gtk_widget_class_set_css_name (widget_class, "phosh-cell-broadcast-prompt");
diff --git a/src/ui/cell-broadcast-prompt.ui b/src/ui/cell-broadcast-prompt.ui
index 1c89d7276..4ddfe0278 100644
--- a/src/ui/cell-broadcast-prompt.ui
+++ b/src/ui/cell-broadcast-prompt.ui
@@ -2,6 +2,7 @@
 <interface>
   <requires lib="gtk+" version="3.24"/>
   <template class="PhoshCellBroadcastPrompt" parent="PhoshSystemModalDialog">
+    <signal name="dialog-canceled" handler="on_dialog_canceled"/>
     <child type="phosh-dialog-content">
       <object class="GtkBox">
         <property name="visible">1</property>
-- 
GitLab


From 9f243f7d96f05b1de7aa091c6721f4ca324f914b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Guido=20G=C3=BCnther?= <agx@sigxcpu.org>
Date: Wed, 6 Aug 2025 18:52:02 +0200
Subject: [PATCH 2/2] cell-broadcast-prompt: Strip whitespace
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Some messages (e.g. Vodafone test messages) have excessive
whitespace at the end of the message, strip that.

Signed-off-by: Guido Günther <agx@sigxcpu.org>
Part-of: <https://gitlab.gnome.org/World/Phosh/phosh/-/merge_requests/1753>
(cherry picked from commit 6ad2078632f5d2350d6b5424160e5f82bae7b393)
---
 src/cell-broadcast-prompt.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cell-broadcast-prompt.c b/src/cell-broadcast-prompt.c
index 78ecf04e7..3354ccc33 100644
--- a/src/cell-broadcast-prompt.c
+++ b/src/cell-broadcast-prompt.c
@@ -61,7 +61,8 @@ set_message (PhoshCellBroadcastPrompt *self, const char *message)
   g_clear_pointer (&self->message, g_free);
   self->message = g_strdup (message);
 
-  gtk_label_set_label (self->lbl_msg, message);
+  g_strstrip (self->message);
+  gtk_label_set_label (self->lbl_msg, self->message);
 
   g_object_notify_by_pspec (G_OBJECT (self), props[PROP_MESSAGE]);
 }
-- 
GitLab

